dds/idl/dds_generator.cpp

```diff
@@ -348,9 +348,7 @@ string type_to_default(const std::string& indent, AST_Type* type, const string&
     AST_Enum* enu = dynamic_cast<AST_Enum*>(actual_type);
     UTL_ScopeActiveIterator i(enu, UTL_Scope::IK_decls);
     AST_EnumVal *item = dynamic_cast<AST_EnumVal*>(i.item());
-    char* temp = item->name()->get_string_copy();
-    def_val = temp;
-    delete [] temp;
+    def_val = item->local_name()->get_string();
```

Comment:
@mitza-oci It looks like this is leaving off the "::DDS::" before some constants in dds/DdsDcpsCoreTypeSupportImpl.cpp. I don't have a great feel for the difference between name and local_name. Should I revert this to the safe version of name->get_string_copy() or does this need to look like the "if (use_cxx11)" block just below?

Reply:
`local_name` is the last component of `name`, so changing them would be significant. I think what Adam meant was `def_val = item->name()->get_string();`.

Reply:
Ah, I thought it could be just the simple name of the enumerator.